portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -57,15 +57,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // store each comment in a comment object
+    long numComments = 0;
+    boolean firstComment = true;
+    long commentsAdded = 0;
+
+    // stores each comment in a comment object
+    // for the last commment submitted, stores the number of comments to print
+    // only creates a comment object for the given number of comments to print
     ArrayList<Comment> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      long timestamp = (long) entity.getProperty("timestamp");
-      String userComment = (String) entity.getProperty("stringValue");
-      
-      Comment comment = new Comment(id, userComment, timestamp);
-      comments.add(comment);
+      if(firstComment){
+          numComments = (long) entity.getProperty("numComments");
+      }
+      if(commentsAdded < numComments){
```

Comment:
I'm a little confused by this logic here. we're tracking the number of comment objects we've created and comparing against the total number of comments in the datastore. But considering the for loop should just run out once we run out of comments, why do we need to check this condition? Is there ever going to be a time when commentsAdded >= numComments?

Reply:
I might need to rename that variable or add some better comments. The numComments property is the user input for 'how many previous comments would you like displayed?' The best way I could think of storing this input was adding it as a property of the entity for every comment and then only taking the value from the most recent comment (hence the if statement for the first comment). So in the second if statement I'm comparing the amount of comments added to the number of comments the user asked to be displayed. The logic is clunky, but it seemed easier than sending all of the information to the javascript file. Should I try to come up with a different way to do this or try to document this way better?

Reply:
I started replying to this below but I'll pick it up here. 

Probably the best way to do this would be to have two different endpoints for these different calls. And have the flow go something like this:

If a user presses the button to update number of displayed comments:
  Call the java servlet to return the correct number of comments and update the display

If a user adds a new comment:
  Use javascript to call the java servlet and add a new comment to the datastore, then use the javascript to call another java servlet to get back the number of comments.
 OR use javascript to call the java servlet save off a new comment to the datastore, then purely use javascript to add another comment to the current display (that being the comment that you just saved) thus saving another call to a servlet and making your website more responsive. This option is more complicated though and the savings are minimal so it's probably not worth doing here but it's worth exploring ways to make as few server calls as possible.

There is a lot to consider here so feel free to reach out tomorrow and we can talk through this.

Reply:
Okay so I've added a different button for limiting the number of comments on the page. I made a new form for the number of comments and connected it to a new servlet, and then fetched both the main servlet and the new one in my load comments function. It should be deployed and working.

Reply:
looks good!